plugin-build/src/main/kotlin/io/sentry/android/gradle/SentryPlugin.kt

```diff
@@ -170,23 +175,35 @@ class SentryPlugin : Plugin<Project> {
                         generateUuidTask.flatMap { it.outputDirectory }
                     )
 
-                    // we just hack ourselves into the proguard task's doLast.
-                    transformerTaskProvider?.configure {
-                        it.finalizedBy(uploadSentryProguardMappingsTask)
+                    if (GroovyCompat.isDexguardEnabledForVariant(project, variant.name)) {
+                        // If Dexguard is enabled, we will have to wait for the project to be evaluated
+                        // to be able to let the uploadSentryProguardMappings run after them.
+                        project.afterEvaluate {
+                            project.tasks.named(
```

Comment:
Just thinking if we can also use `transformerTaskProvider` here instead?

Reply:
`transformerTaskProvider` is null at this stage because `getTransformerTask` wasn't able to find the relevant task. Adding `"dexguardApk${variant.name.capitalized}"` and `"dexguardAab${variant.name.capitalized}"` to `getTransformerTask` also doesn't work as:
1. it requires the `project.afterEvaluate {}` block
2. For Dexguard, `transformerTaskProvider` should return two tasks (`dexguardApkDebug` and `dexguardAabDebug`).